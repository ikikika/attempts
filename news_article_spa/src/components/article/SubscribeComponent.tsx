import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Image from "react-bootstrap/Image";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import InputGroup from "react-bootstrap/InputGroup";
import { ChevronRight, Telegram } from "react-bootstrap-icons";
import parse from "html-react-parser";

import subscribeImage from "../../assets/images/newsletter-sub-new-image.png";
import { useAppSelector } from "../../hooks/reduxHooks";
import { RootState } from "../../redux/store";

const styles = {
  subscribeLabel: {
    backgroundColor: "var(--c-primary)",
    color: "var(--c-text-inverse)",
    fontSize: "var(--fs-subscription)",
    fontFamily: "var(--ff-subscription)",
    lineHeight: "var(--lh-subscription)",
  },
  subscribeText: {
    fontSize: "var(--fs-link)",
    fontFamily: "var(--ff-link)",
    lineHeight: "var(--lh-link)",
  },
  subscribeImage: {
    top: -30,
  },
  subscribeInput: {
    fontSize: "var(--fs-text-sm)",
    fontFamily: "var(--ff-text-sm)",
    lineHeight: "var(--lh-text-sm)",
    color: "var(--c-accent-light)",
    borderColor: "var(--c-muted-accent-high)",
  },
  subscribeButton: {
    fontSize: "var(--fs-button-s)",
    fontFamily: "var(--ff-button-s)",
    fontWeight: 500,
    textTransform: "uppercase" as "uppercase",
    color: "var(--c-text-inverse)",
    backgroundColor: "var(--c-primary)",
    borderColor: "var(--c-primary-accent-high)",
  },
  subscribeDisclaimer: {
    fontSize: "var(--fs-sub-description-s)",
    fontFamily: "var(--ff-sub-description-s)",
    lineHeight: "var(--lh-sub-description-s)",
    color: "var(--c-text)",
  },
  telegramWrapper: {
    backgroundColor: "var(--c-muted)",
  },
  telegramIcon: {
    backgroundColor: "#fff",
    width: 80,
    height: 80,
  },
  telegramCtaStyle1: {
    fontSize: "var(--fs-subscription)",
    fontWeight: 400,
    fontFamily: "var(--ff-subscription)",
    lineHeight: "var(--lh-subscription)",
  },
  telegramCtaStyle2: {
    fontSize: "var(--fs-sub-description)",
    fontWeight: 200,
    fontFamily: "var(--ff-sub-description)",
    lineHeight: "var(--lh-sub-description)",
  },
  redText: {
    color: "var(--c-primary)",
  },
};

const SubscribeComponent = () => {
  const article = useAppSelector(
    (state: RootState) => state.article.data.subscriptionBoxData
  );

  return (
    <Row>
      <Col sm={7} xl={6}>
        <div className="d-flex flex-column border border-1">
          <div className="py-3 ps-4" style={styles.subscribeLabel}>
            <div className="w-75">{article.title}</div>
          </div>

          <Row
            className="px-4 pt-4 pb-md-0 pb-md-4 pb-lg-0 pb-xxl-4 d-flex flex-row"
            style={styles.subscribeText}
          >
            <Col xs={7} xl={6} className="pe-1">
              {parse(article.body ? article.body : "")}
            </Col>
            <Col xs={5} xl={6} className="position-relative">
              <Image
                src={subscribeImage}
                fluid
                className="position-absolute w-100 px-3"
                style={styles.subscribeImage}
              />
            </Col>
          </Row>

          <Row>
            <Col>
              <InputGroup className="p-3 pb-3 pt-xl-3">
                <Form.Control
                  placeholder={article.placeholder}
                  aria-label={article.placeholder}
                  aria-describedby="subscribe-input-field"
                  style={styles.subscribeInput}
                  className="border border-top-0 border-end-0 border-start-0 rounded-0"
                />
                <Button
                  variant="outline-secondary"
                  id="subscribe-input-field"
                  className="rounded-0 border-0 px-3 py-2 text-uppercase"
                  style={styles.subscribeButton}
                >
                  {article.label} <ChevronRight />
                </Button>
              </InputGroup>
            </Col>
          </Row>
          <div className="mt-1 px-3 pb-3" style={styles.subscribeDisclaimer}>
            {article.sub_description}
          </div>
        </div>
      </Col>

      <Col sm={5} xl={6}>
        <a
          href="https://t.me/todayonlinesg"
          className="p-5 p-lg-0 d-flex flex-column justify-content-center align-items-center h-100"
          style={styles.telegramWrapper}
        >
          <div
            className="rounded-circle d-flex justify-content-center align-items-center"
            style={styles.telegramIcon}
          >
            <Telegram size={80} color="#0088CC" />
          </div>
          <div className="my-3" style={styles.telegramCtaStyle1}>
            Follow <span style={styles.redText}>today</span>
            <br />
            on Telegram!
          </div>
          <div style={styles.telegramCtaStyle2}>
            Join our channel at
            <br />
            <span style={styles.redText}>t.me/todayonlinesg</span>
          </div>
        </a>
      </Col>
    </Row>
  );
};

export default SubscribeComponent;
