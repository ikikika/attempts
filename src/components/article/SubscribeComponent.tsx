import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Image from "react-bootstrap/Image";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import InputGroup from "react-bootstrap/InputGroup";
import { ChevronRight, Telegram } from "react-bootstrap-icons";

import subscribeImage from "../../assets/images/newsletter-sub-new-image.png";

const styles = {
  subscribeLabel: {
    backgroundColor: "var(--c-primary)",
    color: "var(--c-text-inverse)",
    fontSize: "var(--fs-subscription)",
    fontFamily: "var(--ff-subscription)",
    lineHeight: "var(--lh-subscription)",
  },
  subscribeText: {
    fontSize: "var(--fs-link)",
    fontFamily: "var(--ff-link)",
    lineHeight: "var(--lh-link)",
  },
  subscribeImage: {
    top: -30,
  },
  subscribeInput: {
    fontSize: "var(--fs-text-sm)",
    fontFamily: "var(--ff-text-sm)",
    lineHeight: "var(--lh-text-sm)",
    color: "var(--c-accent-light)",
    borderColor: "var(--c-muted-accent-high)",
  },
  subscribeButton: {
    fontSize: "var(--fs-button-s)",
    fontFamily: "var(--ff-button-s)",
    fontWeight: 500,
    textTransform: "uppercase" as "uppercase",
    color: "var(--c-text-inverse)",
    backgroundColor: "var(--c-primary)",
    borderColor: "var(--c-primary-accent-high)",
  },
  subscribeDisclaimer: {
    fontSize: "var(--fs-sub-description-s)",
    fontFamily: "var(--ff-sub-description-s)",
    lineHeight: "var(--lh-sub-description-s)",
    color: "var(--c-text)",
  },
  telegramWrapper: {
    backgroundColor: "var(--c-muted)",
  },
  telegramIcon: {
    backgroundColor: "#fff",
    width: 80,
    height: 80,
  },
  telegramCtaStyle1: {
    fontSize: "var(--fs-subscription)",
    fontWeight: 400,
    fontFamily: "var(--ff-subscription)",
    lineHeight: "var(--lh-subscription)",
  },
  telegramCtaStyle2: {
    fontSize: "var(--fs-sub-description)",
    fontWeight: 200,
    fontFamily: "var(--ff-sub-description)",
    lineHeight: "var(--lh-sub-description)",
  },
  redText:{
    color: 'var(--c-primary)'
  }
};

const SubscribeComponent = () => {
  return (
    <Row>
      <Col sm={7} xl={6}>
        <div className="d-flex flex-column border border-1">
          <div className="py-3 ps-4" style={styles.subscribeLabel}>
            Stay in the know. Anytime. <br />
            Anywhere.
          </div>

          <Row
            className="px-4 pt-4 pb-md-0 pb-md-4 pb-lg-0 pb-xxl-4 d-flex flex-row"
            style={styles.subscribeText}
          >
            <Col xs={7} xl={6} className="pe-1">
              Subscribe to get daily news updates, insights and must reads
              delivered straight to your inbox.
            </Col>
            <Col xs={5} xl={6} className="position-relative">
              <Image
                src={subscribeImage}
                fluid
                className="position-absolute w-100 px-3"
                style={styles.subscribeImage}
              />
            </Col>
          </Row>

          <Row>
            <Col>
              <InputGroup className="p-3 pb-3 pt-xl-3">
                <Form.Control
                  placeholder="Enter your email address"
                  aria-label="Enter your email address"
                  aria-describedby="subscribe-input-field"
                  style={styles.subscribeInput}
                  className="border border-top-0 border-end-0 border-start-0 rounded-0"
                />
                <Button
                  variant="outline-secondary"
                  id="subscribe-input-field"
                  className="rounded-0 border-0 px-3 py-2"
                  style={styles.subscribeButton}
                >
                  SUBSCRIBE <ChevronRight />
                </Button>
              </InputGroup>
            </Col>
          </Row>
          <div className="mt-1 px-3 pb-3" style={styles.subscribeDisclaimer}>
            By clicking subscribe, I agree for my personal data to be used to
            send me TODAY newsletters, promotional offers and for research and
            analysis.
          </div>
        </div>
      </Col>

      <Col sm={5} xl={6}>
        <a
          href="https://t.me/todayonlinesg"
          className="p-5 p-lg-0 d-flex flex-column justify-content-center align-items-center h-100"
          style={styles.telegramWrapper}
        >
          <div
            className="rounded-circle d-flex justify-content-center align-items-center"
            style={styles.telegramIcon}
          >
            <Telegram size={80} color="#0088CC" />
          </div>
          <div className="my-3" style={styles.telegramCtaStyle1}>
            Follow <span style={styles.redText}>today</span>
            <br />
            on Telegram!
          </div>
          <div style={styles.telegramCtaStyle2}>
            Join our channel at
            <br />
            <span style={styles.redText}>t.me/todayonlinesg</span>
          </div>
        </a>
      </Col>
    </Row>
  );
};

export default SubscribeComponent;
