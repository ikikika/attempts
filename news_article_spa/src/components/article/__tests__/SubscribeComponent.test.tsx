import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import SubscribeComponent from "../SubscribeComponent";

it("SubscribeComponent matches snapshot", () => {
  const initialState = {
    article: {
      data: {
        subscriptionBoxData: {
          body: "lorem",
          placeholder: "ipsum",
          sub_description: "dolor",
          title: "sit",
          label: "amet",
        },
      },
    },
  };
  const mockStore = configureStore();
  let store = mockStore(initialState);

  const tree = renderer
    .create(
      <Provider store={store}>
        <SubscribeComponent />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
