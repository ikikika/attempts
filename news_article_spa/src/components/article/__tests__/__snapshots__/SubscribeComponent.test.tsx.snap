// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeComponent matches snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xl-6 col-sm-7"
  >
    <div
      className="d-flex flex-column border border-1"
    >
      <div
        className="py-3 ps-4"
        style={
          Object {
            "backgroundColor": "var(--c-primary)",
            "color": "var(--c-text-inverse)",
            "fontFamily": "var(--ff-subscription)",
            "fontSize": "var(--fs-subscription)",
            "lineHeight": "var(--lh-subscription)",
          }
        }
      >
        <div
          className="w-75"
        >
          sit
        </div>
      </div>
      <div
        className="px-4 pt-4 pb-md-0 pb-md-4 pb-lg-0 pb-xxl-4 d-flex flex-row row"
        style={
          Object {
            "fontFamily": "var(--ff-link)",
            "fontSize": "var(--fs-link)",
            "lineHeight": "var(--lh-link)",
          }
        }
      >
        <div
          className="pe-1 col-xl-6 col-7"
        >
          lorem
        </div>
        <div
          className="position-relative col-xl-6 col-5"
        >
          <img
            className="position-absolute w-100 px-3 img-fluid"
            src="newsletter-sub-new-image.png"
            style={
              Object {
                "top": -30,
              }
            }
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="p-3 pb-3 pt-xl-3 input-group"
          >
            <input
              aria-describedby="subscribe-input-field"
              aria-label="ipsum"
              className="border border-top-0 border-end-0 border-start-0 rounded-0 form-control"
              placeholder="ipsum"
              style={
                Object {
                  "borderColor": "var(--c-muted-accent-high)",
                  "color": "var(--c-accent-light)",
                  "fontFamily": "var(--ff-text-sm)",
                  "fontSize": "var(--fs-text-sm)",
                  "lineHeight": "var(--lh-text-sm)",
                }
              }
            />
            <button
              className="rounded-0 border-0 px-3 py-2 text-uppercase btn btn-outline-secondary"
              disabled={false}
              id="subscribe-input-field"
              style={
                Object {
                  "backgroundColor": "var(--c-primary)",
                  "borderColor": "var(--c-primary-accent-high)",
                  "color": "var(--c-text-inverse)",
                  "fontFamily": "var(--ff-button-s)",
                  "fontSize": "var(--fs-button-s)",
                  "fontWeight": 500,
                  "textTransform": "uppercase",
                }
              }
              type="button"
            >
              amet
               
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        className="mt-1 px-3 pb-3"
        style={
          Object {
            "color": "var(--c-text)",
            "fontFamily": "var(--ff-sub-description-s)",
            "fontSize": "var(--fs-sub-description-s)",
            "lineHeight": "var(--lh-sub-description-s)",
          }
        }
      >
        dolor
      </div>
    </div>
  </div>
  <div
    className="col-xl-6 col-sm-5"
  >
    <a
      className="p-5 p-lg-0 d-flex flex-column justify-content-center align-items-center h-100"
      href="https://t.me/todayonlinesg"
      style={
        Object {
          "backgroundColor": "var(--c-muted)",
        }
      }
    >
      <div
        className="rounded-circle d-flex justify-content-center align-items-center"
        style={
          Object {
            "backgroundColor": "#fff",
            "height": 80,
            "width": 80,
          }
        }
      >
        <svg
          fill="#0088CC"
          height={80}
          viewBox="0 0 16 16"
          width={80}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8.287 5.906c-.778.324-2.334.994-4.666 2.01-.378.15-.577.298-.595.442-.03.243.275.339.69.47l.175.055c.408.133.958.288 1.243.294.26.006.549-.1.868-.32 2.179-1.471 3.304-2.214 3.374-2.23.05-.012.12-.026.166.016.047.041.042.12.037.141-.03.129-1.227 1.241-1.846 1.817-.193.18-.33.307-.358.336a8.154 8.154 0 0 1-.188.186c-.38.366-.664.64.015 1.088.327.216.589.393.85.571.284.194.568.387.936.629.093.06.183.125.27.187.331.236.63.448.997.414.214-.02.435-.22.547-.82.265-1.417.786-4.486.906-5.751a1.426 1.426 0 0 0-.013-.315.337.337 0 0 0-.114-.217.526.526 0 0 0-.31-.093c-.3.005-.763.166-2.984 1.09z"
          />
        </svg>
      </div>
      <div
        className="my-3"
        style={
          Object {
            "fontFamily": "var(--ff-subscription)",
            "fontSize": "var(--fs-subscription)",
            "fontWeight": 400,
            "lineHeight": "var(--lh-subscription)",
          }
        }
      >
        Follow 
        <span
          style={
            Object {
              "color": "var(--c-primary)",
            }
          }
        >
          today
        </span>
        <br />
        on Telegram!
      </div>
      <div
        style={
          Object {
            "fontFamily": "var(--ff-sub-description)",
            "fontSize": "var(--fs-sub-description)",
            "fontWeight": 200,
            "lineHeight": "var(--lh-sub-description)",
          }
        }
      >
        Join our channel at
        <br />
        <span
          style={
            Object {
              "color": "var(--c-primary)",
            }
          }
        >
          t.me/todayonlinesg
        </span>
      </div>
    </a>
  </div>
</div>
`;
